concurrent/shared/src/main/scala/zio/concurrent/MVar.scala

```diff
@@ -58,8 +58,7 @@ final class MVar[A] private (private val content: TRef[Option[A]]) {
 
   /**
    * Take a value from an `MVar`, put a new value into the `MVar` and return the
-   * value taken. This function is atomic only if there are no other producers
-   * for this `MVar`.
+   * value taken.
```

Comment:
As far as I know, our implementation of the `swap` function differs from the Haskell. We used `STM` operations that are atomic. Please correct me if I'm wrong.